### DISCUSSION TITLE
Shaping of routes

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#43] [Q&A] [answered]
There is the website [gpxshaping.de](https://gpxshaping.de/GPX/GpxShaping.html) to help "shaping" the routes.

---

_Is there a feature like this that you would like **inside Cruiser** and what exactly?_

e.g. enter more shaping points automatically based on some criteria

**Comment by 0709wiwiwi** [comment, marked as answer]
Wolfgang's idea to add shaping points at regular distances into an empty normal import track in order so to be able to produce a reliable navigation track trajectory as faithfully as possible seems to me to be something worth considering.

Wolfgang's proposal to eventually add the distance to the finish notation in the Shaping point name in addition to the on map shown normal serial number also seems attractive to me. (Max total shaping + via points in a navtrack to calculate = 200 or preferably even less)

Don't rush into this, but think about it very carefully and don't in a hurry add new or change excisting tools and things.
Anyway a slightly more efficient addition of the shaping points to an imported empty track is sure still welcome, imo.
Tip: **Always show the original import track as an overlay reference so to notice eventually unexpected trajectory changes !**

The chance thereby that you generate a false U-turn is still there, but you can quickly notice these now by means of the red U-turn indicator in the turnlist. (GH offline routing)

When using the Beta version of the (offline) Brouter app and provided to add the correct text assign line in the route profile, even such unwanted harmful u-turns are mostly corrected, this is by tests shown to be quite efficiently done by the  autocorrection assign rule.

**Comment by WolfgangZ42** [comment]
Since about 2 years I use _gpxshaping.de_ to add additional shaping points into my kurviger based route for the Garmin/TomTom co-drivers. With this we drive the same route despite different algorithms. 

Now with the switch to Cruiser (and thus without access to the Kurviger server) I also pimp my kurviger file with it to minimize the influence of BRouter during the rare recalculations. 

If Cruiser could automatically fill larger "gaps" with shaping points - e.g. a waypoint at least every 10 km - my workflow could be simplified.

**Comment by MotoUKRider** [comment]
Would this be something to keep you on a more fun "secondary" road that happens to run parallel to a "motorway" and preventing the navigation from putting you onto the faster road at the next available junction?

Incredibly poor artist's impression of minor road with shaping points *after* each exit to ensure route remains on it. Final junction shows what happens if you miss a shaping point.  When planning my routes, I ritually place a shaping point *just after* a significant turn which works 99% of the time for me.

![routing-points](https://user-images.githubusercontent.com/125696109/219946070-20f05911-f988-4aba-b06e-ac2c62f19def.jpg)

**Comment by devemux86** [comment]
> add the distance to the finish notation in the Shaping point name in addition to the on map shown normal serial number

Cruiser shows the distance + time in **via points**:
- From the previous via point and to the next via point
- From the route start and to the route destination

They are displayed in info bubbles on the map and in the waypoints list window.
Also navigation can show the distance and time to a **selected** next via point.

<img src="https://user-images.githubusercontent.com/3484020/219947237-6cfe83f5-aac8-484f-ab19-20069dae9f13.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/219947242-98649ca3-95c9-4204-a01f-68b7298a480d.jpg" width="50%"/>

**Reply by WolfgangZ42** [reply]
**That's true and that's great - but just for us Cruiser users.** 
The "ordinary" Garmin/TomTom user only sees "Pekelne Doly" (nice place btw) ... 
The idea after the distance to finish in waypoint names is to improve the communication between different navigation systems. So after been totally lost - should never occur :smirk: - you can easier decide which is the next valid waypoint (e.g. "56,4 km Pekelne Doly") and give this information to the current leader of the gang, who has a Garmin XT. Because all your waypoint names automatically are sorted you can pick it very quickly. :thumbsup:

**Reply by devemux86** [reply]
I just mentioned it, because in the past we showed distance / time between shaping points too,
then we decided that it was too much and we only need distance / time between the via points.

**Reply by 0709wiwiwi** [reply]
> **The idea after the distance to finish in waypoint names is to improve the communication between different navigation systems.**

- Exactly.   

"To know".

By that Cruiser again so is usefully used as the 'handyman' app to catch shortcomings of commercial GPS units as best as possible.  

_Because of the 100% faithful **TRUE track turn by turn navigation** offered by Cruiser the Android app users are not faced with unexpected route changes. Nevertheless, the Cruiser navigation remains perfectly flexible.  If desired, you can reroute and reuse the integrated Shaping and Via **trackpoints**.  Imagine (dream) the simple to operate Cruiser **navtrack** mode is combined with the available very robust motorbike gps hardware.  ;-)_  Top gear.
